backend/Models/user.model.js

```diff
@@ -1,38 +1,98 @@
-const mongoose = require('mongoose');
+const mongoose = require("mongoose");
 
 const Schema = mongoose.Schema;
 
 // Building user schema and defining the fields
-const userSchema = new Schema({
+const userSchema = new Schema(
+  {
     userID: {
-        type: String,
-        unique: true,
-        required: true,
-        trim: true
+      type: String,
+      unique: true,
+      required: true,
+      trim: true,
     },
     refreshToken: {
-        type: String,
-        required: true,
+      type: String,
+      required: true,
     },
     name: {
-        type: String,
+      type: String,
     },
     imageURL: {
-        type: String
+      type: String,
     },
     email: {
-        type: String
+      type: String,
     },
-    groups: [
-        {
-            type: String
+
+    topTracks: [
+      {
+        songName: {
+          type: String,
+          required: true,
+        },
+        songID: {
+          type: String,
+          required: true,
+        },
+        songPopularity: {
```

Comment:
theoretically, for the pure purpose of displaying top tracks, this would be unnecessary. No harm in having it for now though.

Reply:
maybe remove the required but its semantics

Reply:
imo its better to have these fields required so that the user returns to the frontend are standardized and always include expected fields